airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/upgrade_cdk/pipeline.py

```diff
@@ -0,0 +1,93 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import re
+
+from dagger import Container
+from pipelines.airbyte_ci.connectors.context import ConnectorContext
+from pipelines.airbyte_ci.connectors.reports import ConnectorReport
+from pipelines.helpers import git
+from pipelines.models.steps import Step, StepResult, StepStatus
+
+
+class SetCDKVersion(Step):
+    title = "Set CDK Version"
+
+    def __init__(
+        self,
+        context: ConnectorContext,
+        repo_dir: Container,
+        new_version: str,
+    ):
+        super().__init__(context)
+        self.repo_dir = repo_dir
+        self.new_version = new_version
+
+    async def _run(self) -> StepResult:
+        context: ConnectorContext = self.context
+        setup_py_path = context.connector.code_directory / "setup.py"
+        if not setup_py_path.exists():
+            return StepResult(
+                self,
+                StepStatus.SKIPPED,
+                stdout="Connector does not have a setup.py file.",
+                output_artifact=self.repo_dir,
+            )
+        try:
+            updated_setup_py = self.update_cdk_version(setup_py_path.read_text())
+        except Exception as e:
+            return StepResult(
+                self,
+                StepStatus.FAILURE,
+                stdout=f"Could not add changelog entry: {e}",
+                output_artifact=self.container_with_airbyte_repo,
+            )
+        updated_repo_dir = self.repo_dir.with_new_file(str(setup_py_path), updated_setup_py)
+        return StepResult(
+            self,
+            StepStatus.SUCCESS,
+            stdout=f"Updated CDK version to {self.new_version}",
+            output_artifact=updated_repo_dir,
+        )
```

Comment:
Direct interaction with Dagger primitives (Directory, and File) would decouple this Step logic from the underlying repo structure and allow a simpler maintenance in the future when `airbyte-ci` would target any repo.
```suggestion
    async def _run(self) -> StepResult:
        context: ConnectorContext = self.context
        setup_py_path = context.connector.code_directory / "setup.py"
        if not await context.get_connector_dir(include=["setup.py"]).entries():
            return StepResult(
                self,
                StepStatus.SKIPPED,
                stdout="Connector does not have a setup.py file.",
                output_artifact=self.repo_dir,
            )
        setup_py = context.get_connector_dir(include=["setup.py"]).file("setup.py")
        setup_py_content = await setup_py.contents()
        try:
            updated_setup_py = self.update_cdk_version(setup_py_content)
            updated_connector_dir = (await context.get_connector_dir()).with_new_file("setup.py", updated_setup_py)
            return StepResult(
                self,
                StepStatus.SUCCESS,
                stdout=f"Updated CDK version to {self.new_version}",
                output_artifact=updated_repo_dir,
            )
        except Exception as e:
            return StepResult(
                self,
                StepStatus.FAILURE,
                stderr=f"Could not add changelog entry: {e}",
                exc_info=e,
            )

```